src/main/kotlin/com/sourcegraph/cody/listeners/CodyCaretListener.kt

```diff
@@ -0,0 +1,38 @@
+package com.sourcegraph.cody.listeners
+
+import com.intellij.openapi.command.CommandProcessor
+import com.intellij.openapi.editor.event.CaretEvent
+import com.intellij.openapi.editor.event.CaretListener
+import com.sourcegraph.cody.agent.CodyAgent
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.protocol.ProtocolTextDocument
+import com.sourcegraph.cody.autocomplete.CodyAutocompleteManager
+import com.sourcegraph.cody.vscode.InlineCompletionTriggerKind
+import com.sourcegraph.config.ConfigUtil
+import com.sourcegraph.utils.CodyEditorUtil
+
+class CodyCaretListener : CaretListener {
+  override fun caretPositionChanged(e: CaretEvent) {
+    if (!ConfigUtil.isCodyEnabled()) {
+      return
+    }
+
+    val commandName = CommandProcessor.getInstance().currentCommandName
+    if (commandName == CodyEditorUtil.VIM_EXIT_INSERT_MODE_ACTION) {
+      return
+    }
+
+    val editor = e.editor
+    val project = editor.project
+    val textDocument = ProtocolTextDocument.fromEditor(editor)
+    if (project != null && textDocument != null) {
+      CodyAgentService.withAgent(project) { agent: CodyAgent ->
+        agent.server.textDocumentDidFocus(textDocument)
+      }
+    }
+
+    CodyAutocompleteManager.instance.clearAutocompleteSuggestions(e.editor)
+    CodyAutocompleteManager.instance.triggerAutocomplete(
```

Comment:
Surprised that IntelliJ platform doesn't handle this. Isn't there[ a "show suggestions as you type" setting](https://www.jetbrains.com/help/idea/auto-completing-code.html) where you can turn off automatic autocomplete and rely on manual triggers? Does this call honor that setting?

Reply:
That is Cody completion, which is completely separate than IntelliJ suggestions and have different UX as well (e.g. can have multiline suggestions). It have also it's own setting to enable/disable it. I do not think we want to mix those two.

Or do I misunderstood your question?

Reply:
> That is Cody completion, which is completely separate than IntelliJ suggestions

Super! Thanks for clarifying.